drivers/driver/driver.go

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2018, Juniper Networks, Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package netconf
+
+import rpc "github.com/arsonistgopher/go-netconf/rpc"
+
+// Driver interface for building drivers that are self-contained from a user's perspective.
+type Driver interface {
+	Lock(ds string) (*rpc.RPCReply, error)
+	Unlock(ds string) (*rpc.RPCReply, error)
```

Comment:
The methods `Lock`,  `Unlock`, `SendRaw`,  `GetConfig` are equal independent of the driver (e.g. ssh.go, junos.go, ...) and should be therefore not part of the Driver interface.